doc/flame/components.md

```diff
@@ -535,104 +554,165 @@ It is also possible to create custom renderers by extending the `ParallaxRendere
+
+In the image you can see how the polygon shape formed by the purple arrows is defined by the red
+arrows.
+
+Remember to define the lists in a counter clockwise manner (if you think in the screen coordinate
+system where the y-axis is flipped, otherwise it is clockwise).
+
+
 ### RectangleComponent
-A `RectangleComponent` can be created in two ways, depending on if it's a square or not.
-To create a `RectangleComponent` that is 300 in width and 200 in height you can do the following:
 
-Example:
+A `RectangleComponent` is created very similarly to how a `PositionComponent` is created, since it
+also has a bounding rectangle.
+
+Something like this for example:
+
 ```dart
-final paint = BasicPalette.red.paint()..style = PaintingStyle.stroke;
-final rectangle = RectangleComponent(
-  size: Vector2(300.0, 200.0),
-  position: Vector2(100, 200),
-  paint: paint,
-);
+void main() {
+  RectangleComponent(
+    position: Vector2(10.0, 15.0),
+    size: Vector2.all(10),
+    angle: pi/2,
+    anchor: Anchor.center,
+  );
+}
 ```
 
-To create a square you can instead use the slightly simpler named constructor
-`RectangleComponent.square`. This is an example of how to create a red square with width and height
-200:
+Dart also already has an excellent way to create rectangles and that class is called `Rect`, you can
+create a Flame `RectangleComponent` from a `Rect` by using the `Rectangle.fromRect` factory, and
+just like when setting the vertices of the `PolygonComponent`, your rectangle will be sized
+according to the `Rect` if you use this constructor.
+
+The following would create a `RectangleComponent` with its top left corner in `(10, 10)` and a size
+of `(100, 50)`.
 
 ```dart
-final paint = BasicPalette.red.paint()..style = PaintingStyle.stroke;
-final square = RectangleComponent.square(
-  size: 200.0,
-  position: Vector2(100, 200),
-  paint: paint,
-);
+void main() {
+  RectangleComponent.fromRect(
+    Rect.fromLTWH(10, 10, 100, 50),
+  );
+}
 ```
 
-### PolygonComponent
-The `PolygonComponent` is the most complicated of the `ShapeComponent`s since you'll have to define
-all the "corners" of your polygon. You can create the `PolygonComponent` in two different ways,
-either you use the default constructor which takes a list of `Vector2` where each of them should be
-between -1.0 and 1.0 that describes the ration of the length from the center to the edge of the size
-of the component. So
-`[Vector2(1.0, 1.0), Vector2(1.0, -1.0), Vector2(-1.0, -1.0), Vector2(-1.0, 1.0)]`
-would describe a rectangle that fills the full size of the component. Remember to define the list in
-a counter clockwise manner (if you think in the screen coordinate system where the y-axis is
-flipped, otherwise it is clockwise).
-
-So to create a diamond shaped `PolygonComponent` which is slightly smaller than the defined size you
-would do this:
+You can also create a `RectangleComponent` from a normal, or use the default constructor to build
```

Comment:
What does "normal" mean here?

Reply:
True, that has to be explained more. It is the point but in relation to the parent size instead of in an absolute position.

Reply:
Fixed

Reply:
But why is it called "normal"? I only heard of "normal" meaning "standard" or "orthogonal", neither of which seem to fit here.

Reply:
I found that name fitting when researching it earlier, but I can't find out where I got it from again and I agree it isn't a very good name. Do you have any suggestions for the name?

Reply:
Perhaps "relative"? I'm not sure what the intended use-case for this constructor, though.

Reply:
It is used to create a hitbox which size is relative to the intended parent's size, it was quite useful in the last collision detection system. 

Now I remember where `normal` came from, it came from that you can normalize a `Vector2`. Still a really bad name though since the normal of a vector is something completely different than a normalized vector. I'll change it to something else, maybe relative.

Reply:
The named constructors are now called `.relative` and the argument that refines the relation is called `relation`, wdyt?

Reply:
:+1: